### PR TITLE
nixos/modules: stdenv.lib -> lib

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

Part of #108938

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/searx.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@ import ./make-test-python.nix ({ pkgs, ...} :
 
 {
   name = "searx";
-  meta = with pkgs.stdenv.lib.maintainers; {
+  meta = with pkgs.lib.maintainers; {
```
Comment (rnhmjoj) [inline nixos/tests/searx.nix:5]: Should we also do `pkgs.lib` -> `lib` in NixOS modules?

Comment (prusnak) [reply nixos/tests/searx.nix:5]: Not sure, but I prefer not to do this in this commit anyway.

**Review by rnhmjoj — APPROVED** [review]
Diff looks good and it evaluates.

File: nixos/modules/config/gnu.nix [hunk 2]
```diff
@@ -1,11 +1,9 @@
 { config, lib, pkgs, ... }:
 
-with lib;
```
Comment (roberth) [inline nixos/modules/config/gnu.nix:3]: Good riddance!
:heart:

**Review by roberth — APPROVED** [review]

**Pull request merged**